File: graphql-dgs-codegen-core/src/integTest/java/com/netflix/graphql/dgs/codegen/java/testcases/dataclasses/dataClassWithDeeplyNestedComplexField/expected/client/PerformanceProjection.java
package com.netflix.graphql.dgs.codegen.java.testcases.dataclasses.dataClassWithDeeplyNestedComplexField.expected.client;

import com.netflix.graphql.dgs.client.codegen.BaseSubProjectionNode;

public class PerformanceProjection<PARENT extends BaseSubProjectionNode<?, ?>, ROOT extends BaseSubProjectionNode<?, ?>> extends BaseSubProjectionNode<PARENT, ROOT> {
  public PerformanceProjection(PARENT parent, ROOT root) {
    super(parent, root, java.util.Optional.of("Performance"));
  }

  public PerformanceProjection<PARENT, ROOT> __typename() {
    getFields().put("__typename", null);
    return this;
  }

  public PerformanceProjection<PARENT, ROOT> zeroToSixty() {
    getFields().put("zeroToSixty", null);
    return this;
  }

  public PerformanceProjection<PARENT, ROOT> quarterMile() {
    getFields().put("quarterMile", null);
    return this;
  }
}

File: graphql-dgs-codegen-core/src/test/resources/kotlin2/dataClassWithDeeplyNestedComplexField/expected/client/PerformanceProjection.kt
package kotlin2.dataClassWithDeeplyNestedComplexField.expected.client

import com.netflix.graphql.dgs.client.codegen.GraphQLProjection

public class PerformanceProjection : GraphQLProjection() {
  public val zeroToSixty: PerformanceProjection
    get() {
      field("zeroToSixty")
      return this
    }

  public val quarterMile: PerformanceProjection
    get() {
      field("quarterMile")
      return this
    }
}
